### PR TITLE
[MODORDERS-841] - Encumbrance current amount is calculated incorrectly after changing fund distribution when related paid invoice exists

### PR DESCRIPTION
**Pull request by Abdulkhakimov**
## Purpose
There was observed that after moving expended value, expended amount didn't recalculate. Changes in the PR should fix this behaviour 
Encumbrance amount calculation formula: (initialAmmountEncumbered - ammountAwaitingPayment - amountExpended)
If result will be less than 0 we should set up amount as 0

## Approach

- Moved recalculation to the separate method
- Replaced using Double to BigDecimal
- Added encumbrance amount calculation formula.
- Updated [karate scenario](https://github.com/folio-org/folio-integration-tests/pull/851)

## Learning
[MODORDERS-841](https://issues.folio.org/browse/MODORDERS-841)

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all the appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?
- Did you modify code to call some additional endpoints?
  - [ ] If so, do you check that necessary module permission added in ModuleDescriptor-template.yaml file?

Ideally, all the PRs involved in breaking changes would be merged on the same day
to avoid breaking the folio-testing environment.
Communication is paramount if that is to be achieved,
especially as the number of inter-module and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems,
ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/service/invoice/POLInvoiceLineRelationService.java [hunk 1]
```diff
@@ -106,6 +104,22 @@ private static org.folio.rest.acq.model.invoice.FundDistribution convertToInvoic
...
+    BigDecimal amountAwaitingPayment = forDelete.stream().map(encumbrance -> encumbrance.getEncumbrance().getAmountAwaitingPayment())
+      .map(BigDecimal::valueOf).reduce(BigDecimal.ZERO, BigDecimal::add);
+
+    forCreate.stream().findFirst().ifPresent(transaction -> {
+      Encumbrance encumbrance = transaction.getEncumbrance();
+      BigDecimal encumbranceAmount = BigDecimal.valueOf(encumbrance.getInitialAmountEncumbered())
+        .subtract(amountExpended).subtract(amountAwaitingPayment).max(BigDecimal.ZERO);
+      transaction.withAmount(encumbranceAmount.doubleValue())
+        .withEncumbrance(encumbrance.withAmountExpended(amountExpended.doubleValue()).withAmountAwaitingPayment(amountAwaitingPayment.doubleValue())
+          .withStatus(encumbranceAmount.compareTo(BigDecimal.ZERO) == 0 ? Encumbrance.Status.RELEASED : Encumbrance.Status.UNRELEASED));
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/service/invoice/POLInvoiceLineRelationService.java:119]: Do we have these formulas in some common place to reuse it?

Comment (Abdulkhakimov) [reply src/main/java/org/folio/service/invoice/POLInvoiceLineRelationService.java]: Good point. Executed to separate method. Thank you!

**Review by SerhiiNosko — APPROVED** [review]

**Review by grabsefx — APPROVED** [review]

**Comment by okolawole-ebsco** [comment]
any tests possible for this? here or ITs?

**Comment by Abdulkhakimov** [comment]
> > any tests possible for this? here or ITs?
> 
> Yes, integration test was created in the same time when this PR is merged: [folio-org/folio-integration-tests#851](https://github.com/folio-org/folio-integration-tests/pull/851)
> 
> Forget to add this integration test PR to description of this PR. Will add in description in future

Hi @okolawole-ebsco @SerhiiNosko  a link is specified in the PR description:
image

**Pull request merged**